cmds/core/touch/access_linux_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func TestAccess(t *testing.T) {
+	tmp := t.TempDir()
+	f, err := os.CreateTemp(tmp, "touch_test")
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	accessDate, err := time.Parse(time.RFC3339, "2023-01-01T00:00:00Z")
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	cmd := command(nil, params{time: accessDate, access: true}, f.Name())
+	err = cmd.run()
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	mfi, err := f.Stat()
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	expected := accessDate.UnixNano()
+	at := mfi.Sys().(*syscall.Stat_t).Atim.Nano()
```

Comment:
Would it be safer to do:
at, ok := mfi.Sys().(*syscall.Stat_t)
and test ok? I worry a bit that mf.Sys() might at some point change -- it being of type any, I wonder how much that is guaranteed.

Reply:
It would be a big change, now it returns a pointer to syscall.Stat_t, so even if underlying Stat_t will change we probably will be able to fetch access time. I think in case of test it should be okay and it will run only under Linux.